src/Http/Action/Security/ShowLogInAction.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Copyright (c) 2013-2018 OpenCFP
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ *
+ * @see https://github.com/opencfp/opencfp
+ */
+
+namespace OpenCFP\Http\Action\Security;
+
+use OpenCFP\Domain\Services;
+use Symfony\Component\HttpFoundation;
+use Symfony\Component\Routing;
+use Twig_Environment;
+
+final class ShowLogInAction
+{
+    /**
+     * @var Services\Authentication
+     */
+    private $authentication;
+
+    /**
+     * @var Twig_Environment
+     */
+    private $twig;
+
+    /** @var string */
+    private $sso;
+
+    /**
+     * @var Routing\Generator\UrlGeneratorInterface
+     */
+    private $urlGenerator;
+
+    public function __construct(
+        Services\Authentication $authentication,
+        Twig_Environment $twig,
+        Routing\Generator\UrlGeneratorInterface $urlGenerator,
+        string $sso
+    ) {
+        $this->authentication = $authentication;
+        $this->twig           = $twig;
+        $this->urlGenerator   = $urlGenerator;
+        $this->sso            = $sso;
+    }
+
+    public function __invoke(HttpFoundation\Request $request): HttpFoundation\Response
+    {
+        /**
+         * If we're already logged in, redirect the user to the dashboard
+         * Otherwise show the login page
+         */
+        try {
+            $this->authentication->authenticate(
+                $request->get('email'),
+                $request->get('password')
+            );
+            $url = $this->urlGenerator->generate('dashboard');
+
+            return new HttpFoundation\RedirectResponse($url);
```

Comment:
Can we move this and the previous statement to after the `try`-and-`catch` block?

I suppose the only thing that can go wrong here and that we care about is the authentication.

Reply:
I don't see a reason to move it. Convince me!

Reply:
None of

```php
$url = $this->urlGenerator->generate('dashboard');

return new HttpFoundation\RedirectResponse($url);
```

will throw an `AuthenticationException`. 

Instead, these two statements constitute the success case on the happy path. By moving them out of the `try` block we make it more clear that they are on that happy path. 

Still, an exception (albeit not an `AuthenticationException`) might be thrown, perhaps when a URL with the `dashboard` identifier could not be generated, but that is clearly out of the scope of handling the authentication.

Does that work?